packages/ui/src/components/vuestic-components/va-dropdown/VaDropdown.vue

```diff
@@ -147,20 +155,24 @@ export default class VaDropdown extends mixins(DropdownPropsMixin) {
     if (this.disabled) {
       return
     }
-    if (!this.isMouseHovered) {
-      this.hoverOverDebounceLoader.run()
-    }
+    if (this.trigger === 'hover') {
+      if (!this.valueComputed) {
+        this.hoverOverDebounceLoader.run()
+      }
 
-    this.hoverOutDebounceLoader.reset()
+      this.hoverOutDebounceLoader.reset()
+    }
   }
 
   onMouseOut (): void {
-    if (this.isContentHoverable) {
-      this.hoverOutDebounceLoader.run()
-    } else {
-      this.isMouseHovered = false
+    if (this.trigger === 'hover') {
```

Comment:
You may generally want to use [early return](https://codeburst.io/code-js-in-stlye-early-return-3462323e30d1) instead of if/else.

Reply:
Thanks for the info, I've read the article and usually using this approach,

In this case however, if I understand it correctly, we don't need to return because we want to manage onMouseOut event by DebounceLoader functions.

Reply:
Why not?

```ts
// We only care about mouseout for hover trigger
if (this.trigger !== 'hover') {
  return
}

if (this.isContentHoverable) {
  this.hoverOutDebounceLoader.run()
} else {
  this.valueComputed = false
}
this.hoverOverDebounceLoader.reset()
```